src/subdomains/supporting/payment/services/fee.service.ts

```diff
@@ -168,36 +170,45 @@ export class FeeService {
       'rate',
     );
 
-    // get max discount > 0
+    // get max discount
     const positiveDiscountFee = Util.maxObj(
-      fees.filter((fee) => fee.type === FeeType.DISCOUNT && (fee.rate > 0 || (fee.rate == 0 && fee.fixed > 0))),
+      fees.filter((fee) => fee.type === FeeType.DISCOUNT),
       'rate',
     );
 
-    // get min discount < 0
-    const negativeDiscountFee = Util.minObj(
-      fees.filter((fee) => fee.type === FeeType.DISCOUNT && (fee.rate < 0 || (fee.rate == 0 && fee.fixed < 0))),
-      'rate',
-    );
+    // get addition fees
+    const additionFees = fees.filter((fee) => fee.type === FeeType.ADDITION);
 
     const discountFee: FeeDto = {
-      fees: [positiveDiscountFee, negativeDiscountFee],
-      rate: (positiveDiscountFee?.rate ?? 0) + (negativeDiscountFee?.rate ?? 0),
-      fixed: (positiveDiscountFee?.fixed ?? 0) + (negativeDiscountFee?.fixed ?? 0),
-      payoutRefBonus: (positiveDiscountFee?.payoutRefBonus ?? true) && (negativeDiscountFee?.payoutRefBonus ?? true),
+      fees: [positiveDiscountFee, ...additionFees],
+      rate: (positiveDiscountFee?.rate ?? 0) + Util.sumObjValue(additionFees, 'rate'),
+      fixed: (positiveDiscountFee?.fixed ?? 0) + Util.sumObjValue(additionFees, 'fixed'),
+      payoutRefBonus:
+        (positiveDiscountFee?.payoutRefBonus ?? true) && (additionFees?.some((fee) => !fee.payoutRefBonus) ?? true),
```

Comment:
`additionFees.every((fee) => fee.payoutRefBonus)`
additionFees cannot be undefined?